commit 1fec12a560033ebe8fa6857dd3cbf9677371fbee
Author: Josef Bacik <josef@toxicpanda.com>
Date:   Wed Dec 16 11:18:45 2020 -0500

    btrfs: noinline btrfs_should_cancel_balance
    
    I was attempting to reproduce a problem that Zygo hit, but my error
    injection wasn't firing for a few of the common calls to
    btrfs_should_cancel_balance.  This is because the compiler decided to
    inline it at these spots.  Keep this from happening by explicitly
    marking the function as noinline so that error injection will always
    work.
    
    Reviewed-by: Qu Wenruo <wqu@suse.com>
    Reviewed-by: Johannes Thumshirn <johannes.thumshirn@wdc.com>
    Reviewed-by: Nikolay Borisov <nborisov@suse.com>
    Signed-off-by: Josef Bacik <josef@toxicpanda.com>
    Reviewed-by: David Sterba <dsterba@suse.com>
    Signed-off-by: David Sterba <dsterba@suse.com>

diff --git a/fs/btrfs/relocation.c b/fs/btrfs/relocation.c
index 2698805ebd26..8e51b39cbfbb 100644
--- a/fs/btrfs/relocation.c
+++ b/fs/btrfs/relocation.c
@@ -2615,7 +2615,7 @@ int setup_extent_mapping(struct inode *inode, u64 start, u64 end,
 /*
  * Allow error injection to test balance cancellation
  */
-int btrfs_should_cancel_balance(struct btrfs_fs_info *fs_info)
+noinline int btrfs_should_cancel_balance(struct btrfs_fs_info *fs_info)
 {
 	return atomic_read(&fs_info->balance_cancel_req) ||
 		fatal_signal_pending(current);